### ISSUE TITLE
[fastrtps] <debug model crash>

### ISSUE THREAD

**Issue by zhangzhen5729** [microsoft/vcpkg#39954]
### Is your feature request related to a problem? Please describe.

1、First I made sure vcpkg executed git pull and update.
2、Installation screenshot:
![image](https://github.com/user-attachments/assets/f8d797c7-a33f-4f70-9a64-fb9361f74525)
![image](https://github.com/user-attachments/assets/6a41eec2-b7ec-48b9-b66b-8e3deff21401)
3、Cmake project reference method:
find_package(fastcdr REQUIRED)
find_package(fastrtps REQUIRED)
target_link_libraries(${PROJECT_NAME} fastcdr)
target_link_libraries(${PROJECT_NAME} fastrtps)
4、run:
debug model crash
![image](https://github.com/user-attachments/assets/2ff08d3b-3dbe-4932-b00a-103ba6745294)

release model normal

### Proposed solution

Use the .cmake included in the windows platform binary installation package provided by fastdds to introduce it into the project. Any mode is normal.
So I suggest checking the references of vcpkg to the Fastdds library in debug and release modes. Compare the cmake scripts to check if there are any problems with the introduction in debug mode.

### Describe alternatives you've considered

If I directly introduce the .cmake configuration file list in the fatsdds official binary installation package in the cmake project (APPEND CMAKE_PREFIX_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../3rd/fastdds/cmake"), it is easy to use in this case. vcpkg should compare the difference with the official .cmake direct introduction to determine whether there is a reference error problem.
![image](https://github.com/user-attachments/assets/9d913e89-284b-4f9b-850e-2e4b6d6a942c)
The above picture is the cmake configuration file in the official binary installation package

**Comment by Cheney-W** [comment]
What's the result when you use below commands?
```
find_package(fastrtps CONFIG REQUIRED)
target_link_libraries(main PRIVATE fastdds::optionparser)
```

**Comment by zhangzhen5729** [comment]
find_package(fastcdr CONFIG REQUIRED)
find_package(fastrtps CONFIG REQUIRED)
target_link_libraries(${PROJECT_NAME} PRIVATE fastcdr)
target_link_libraries(${PROJECT_NAME} PRIVATE fastdds::optionparser)

result:
![image](https://github.com/user-attachments/assets/a73c4b8c-c251-4db7-ab89-e1e4bdd480cf)
Why transfer static libraries?

**Comment by zhangzhen5729** [comment]
![image](https://github.com/user-attachments/assets/b75f2b3c-b0da-4233-b5b7-53eaa1d97c49)
![image](https://github.com/user-attachments/assets/340ad5ed-67cd-431c-be7a-da16430cdb93)

**Comment by dg0yt** [comment]
> Why transfer static libraries?

There is a `fastrtpsd-2.14.lib` "import lib" for dynamic library linkage. i.e. not a static lib, and no `lib` prefix.

**Comment by zhangzhen5729** [comment]
@dg0yt I used the cmake style find_package and controlled the parameters -Dfastrtps_SHARED_LIBS=ON -Dfastcdr_SHARED_LIBS=ON to reference the dynamic library, but an error was reported saying that libfastrtpsd-2.14.lib could not be found.

**Comment by zhangzhen5729** [comment]
If I first introduce the cmake folder path in the official FASTDDS binary installation package, and then uninstall the fastdds installed by vcpkg, everything will be normal after compilation.

**Comment by dg0yt** [comment]
`-Dfastrtps_SHARED_LIBS=ON -Dfastcdr_SHARED_LIBS=ON` will probably not interact as expected with vcpkg:
The linkage offered by fastrtps in vcpkg is controlled only by the vcpkg triplet and port. `x64-windows` normally means shared library linkage.

Anyways you must check where the unexpected `lib` prefix in `libfastrtpsd-2.14.lib` comes from.
First try a clean build with vpckg, i.e. without any CMake cache.

(Does the "official FASTDDS binary installation" provide binaries with `lib` prefix?)

**Comment by zhangzhen5729** [comment]
yes,provide binaries with prefix
![image](https://github.com/user-attachments/assets/d9fa4925-637f-41fe-9075-9ca6788f988d)

**Comment by zhangzhen5729** [comment]
What you said makes sense. Maybe the parameters -Dfastrtps_SHARED_LIBS=ON -Dfastcdr_SHARED_LIBS=ON are not effective in vcpkg. Next, I will try to delete any cmake cache before using vcpkg.

**Comment by zhangzhen5729** [comment]
![image](https://github.com/user-attachments/assets/6237e6e1-67f1-4e57-9559-d34d2b484c8f)
 -Dfastrtps_SHARED_LIBS=ON -Dfastcdr_SHARED_LIBS=ON How should vcpkg be used to make this parameter effective?

**Comment by dg0yt** [comment]
> -Dfastrtps_SHARED_LIBS=ON -Dfastcdr_SHARED_LIBS=ON How should vcpkg be used to make this parameter effective?

Basically, if these options don't match the vcpkg triplet, the cmake package must return an error.

**Comment by zhangzhen5729** [comment]
I tried againafter clearing the cache  and 
find_package (fastcdr required)
find_package (requires fastrtps) 
target_link_libraries (${PROJECT_NAME} fastcdr)
target_link_libraries(${PROJECT_NAME} fastrtps)
 can be compiled normally using this method, but the debug mode crashes and the release mode is normal;

If add this sentence 
list(APPEND CMAKE_PREFIX_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../3rd/fastdds/cmake")
 to refer to the cmake folder of the official binary installation package, and other statements remain unchanged, any mode can run normally.

**Comment by zhangzhen5729** [comment]
find_package(fastcdr CONFIG REQUIRED)
find_package(fastrtps CONFIG REQUIRED)
target_link_libraries(${PROJECT_NAME} PRIVATE fastcdr)
target_link_libraries(${PROJECT_NAME} PRIVATE fastdds::optionparser)       error: not open libfastrtpsd-2.14.lib

**Comment by dg0yt** [comment]
What I understand:

1. After a clean build *with vcpkg deps only*, debug mode crashes.

> I tried again after clearing the cache and 
> find_package (fastcdr required)
> find_package (requires fastrtps) 
> target_link_libraries (${PROJECT_NAME} fastcdr) 
> target_link_libraries(${PROJECT_NAME} fastrtps) 
can be compiled normally using this method, but the debug mode crashes and the release mode is normal;

2. After changing variables that modify `find_xxx` behavior to search the official distribution, code runs normally.

> If add this sentence 
> list(APPEND CMAKE_PREFIX_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../3rd/fastdds/cmake") 
> to refer to the cmake folder of the official binary installation package, and other statements remain unchanged, any mode can run normally.

3. `fastdds::optionparser` introduces a link reference to a missing `libfastrtpsd-2.14.lib`.

> find_package(fastcdr CONFIG REQUIRED)
> find_package(fastrtps CONFIG REQUIRED)
> target_link_libraries(${PROJECT_NAME} PRIVATE fastcdr) 
> target_link_libraries(${PROJECT_NAME} PRIVATE fastdds::optionparser) 
> error: not open libfastrtpsd-2.14.lib

These might be three different sub-issues. Needs more analyzes.

**Comment by zhangzhen5729** [comment]
Yes, I think as long as the debug mode of question 1 can run normally, it meets my needs
Question 2 is an auxiliary solution to problem 1 in terms of thinking
Question 3 is a different way of referencing question 1

**Comment by zhangzhen5729** [comment]
Yes, I think as long as the debug mode of question 1 can run normally, it meets my needs

Question 2 is an auxiliary solution to problem 1 in terms of thinking

Question 3 is a different way of referencing question 1
